core/controllers/reader.py

```diff
@@ -929,6 +975,9 @@ def get(self, exploration_id):
                 next_exp_id = (
                     collection.get_next_exploration_id_in_sequence(
                         exploration_id))
+        elif story_id and current_node_id:
```

Comment:
This is happening in the wrong place. As a rule, GET operations should never modify state.

Reply:
Done, added a PUT function to this handler.